home/trainers.asm

```diff
@@ -128,14 +128,16 @@ TalkToTrainer::
 ; checks if any trainers are seeing the player and wanting to fight
 CheckFightingMapTrainers::
 IF DEF(_DEBUG)
-	call DebugPressedOrHeldB
+	call DebugBButtonToggle
 	jr nz, .trainerNotEngaging
 ENDC
 	call CheckForEngagingTrainers
 	ld a, [wSpriteIndex]
-	cp $ff
+	cp -1
```

Comment:
`-1` is the same value as `$ff`. If you're going to change all the `$ff`s related to `wSpriteIndex`, it should be because there's some `SPECIAL_SPRITE_INDEX EQU -1` constant being added.

Reply:
> it should be because there's some SPECIAL_SPRITE_INDEX EQU -1 constant being added

Well, should there be such a constant? In any case, I'll revert the change since that would be out of scope for this PR.

Reply:
Yeah, it might be appropriate for a future PR, depending on where the `-1` value comes from which this code checks for.